corehq/apps/analytics/static/analytics/js/google.js

```diff
@@ -0,0 +1,148 @@
+/* globals _, $, Array, window */
+/**
+ *  Handles communication with the google analytics API. gtag is the replacement
+ *  for Google's old analytics.js (ga).
+ */
+hqDefine('analytics/js/google', function () {
+    'use strict';
+    var _get = hqImport('analytics/js/initial').getFn('google'),
+        logger = hqImport('analytics/js/logging').getLoggerForApi('Google Analytics'),
+        _utils = hqImport('analytics/js/utils'),
+        _init = {},
+        _gtag;
+
+    logger.verbose.addCategory('data', 'DATA');
+
+    _init.apiId = _get('apiId');
+    logger.verbose.data(_init.apiId || "NOT SET", "API ID");
+
+    var _addGoogleScript = function (srcUrl) {
+        logger.verbose.log(srcUrl, "Added Script");
+        _utils.insertAsyncScript(srcUrl);
+    };
+
+    if (_init.apiId) {
+        _addGoogleScript('//www.googletagmanager.com/gtag/js?id=' + _init.apiId);
+    }
+
+    window.dataLayer = window.dataLayer || [];
```

Comment:
I don't understand exactly what `window.dataLayer` does - looks like it's storing all of the kiss and GA events, but also like we only push data to it, never read from it...is that right? Why do these need to be stored in the window?

Reply:
`window.dataLayer` is what `gtag` is reading from. There are integrations for google analytics with kissmetrics data now as well.
https://developers.google.com/analytics/devguides/collection/gtagjs/